enable qat for custom annotation in qnn

navsud opened the pull request:
Summary: This diff enables QAT for `annotate_matmul_16a8w` custom annotation. After this change, verified that the prepared graph shows FakeQuants (QAT) instead of Observers (PTQ).

Differential Revision: D79705374

shewu-quic reviewed (approved):
LGTM. Thank you for the efforts.

cccclai reviewed (approved):
Thanks!

cccclai commented:
@navsud can you fix the lint?

navsud committed:
enable qat for custom annotation in qnn (#13147)

Summary:

This diff enables QAT for `annotate_matmul_16a8w` custom annotation. After this change, verified that the prepared graph shows FakeQuants (QAT) instead of Observers (PTQ).

Differential Revision: D79705374
